ipalib/rpc.py

```diff
@@ -118,7 +119,10 @@ def read_persistent_client_session_data(principal):
     '''
 
     try:
-        return session_storage.get_data(principal, CCACHE_COOKIE_KEY)
+        value = session_storage.get_data(principal, CCACHE_COOKIE_KEY)
+        if value == CCACHE_COOKIE_EMPTY_VALUE:
```

Comment:
This is exactly the same code pattern as it was before. It is there for a reason and I don't want to change it.

Python 3 should be returning bytes, so comparison needs to change, yes.